### PR TITLE
[WIP] Add Support to install OCP over BM

### PR DESCRIPTION
**Pull request by prsurve**
Signed-off-by: Pratik Surve <prsurve@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/baremetal.py [hunk 1]
```diff
@@ -1,13 +1,410 @@
...
+            cmd = "systemctl restart dnsmasq"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to restart dnsmasq service"
+            with open(constants.RHCOS_IMAGES_FILE) as file_stream:
+                rhcos_images_file = yaml.safe_load(file_stream)
+            ocp_version = get_ocp_version()
+            logger.info(rhcos_images_file)
+            image_data = rhcos_images_file[ocp_version]
+            # Download installer_initramfs
+            cmd = f"wget -O {constants.bm_tftp_dir}/rhcos-installer-initramfs.x86_64.img {constants.coreos_url_prefix}" \
+                  f"{image_data['installer_initramfs_url']}"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:164]: break lines with parenthesis

File: ocs_ci/deployment/baremetal.py [hunk 2]
```diff
@@ -1,13 +1,410 @@
...
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            # Download installer_kernel
+            cmd = f"wget -O {constants.bm_tftp_dir}/rhcos-installer-kernel-x86_64 {constants.coreos_url_prefix}" \
+                  f"{image_data['installer_kernel_url']}"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:168]: same

File: ocs_ci/deployment/baremetal.py [hunk 3]
```diff
@@ -1,13 +1,410 @@
...
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            # Download metal_bios
+            cmd = f"wget -O {constants.bm_path_to_upload}/rhcos-metal.x86_64.raw.gz {constants.coreos_url_prefix}" \
+                  f"{image_data['metal_bios_url']}"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:172]: same

File: ocs_ci/deployment/baremetal.py [hunk 4]
```diff
@@ -1,13 +1,410 @@
...
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to create required folder"
+
+        def deploy(self, log_cli_level='DEBUG'):
+            """
+            Deploy
+            """
+            # Uploading pxe files
+            logger.info("Deploying OCP cluster for Bare Metal platform")
+            logger.info(
+                f"Openshift-installer will be using loglevel:{log_cli_level}"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:185]: log level

File: ocs_ci/deployment/baremetal.py [hunk 5]
```diff
@@ -1,13 +1,410 @@
...
+                    key_file=self.private_key
+                )
+            # Restarting dnsmasq service
+            cmd = "systemctl restart dnsmasq"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to restart dnsmasq service"
+
+            # Rebooting Machine with pxe boot
+            self.mgmt_details = load_auth_config()['ipmi']
+            for machine_id in range(1, 6):
+                machine_name = f"argo00{1}.ceph.redhat.com"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:216]: please save the machine name template in a conf file

File: ocs_ci/deployment/baremetal.py [hunk 6]
```diff
@@ -1,13 +1,410 @@
...
+                if self.mgmt_details[machine_name]:
+                    # Power off machine
+                    cmd = f"ipmitool -I lanplus -U {self.mgmt_details[machine_name]['mgmt_username']} " \
+                          f"-P {self.mgmt_details[machine_name]['mgmt_password']} " \
+                          f"-H {self.mgmt_details[machine_name]['mgmt_console']} chassis power off"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:221]: break lines with parentheses

File: ocs_ci/deployment/baremetal.py [hunk 7]
```diff
@@ -1,13 +1,410 @@
...
+                    secrets = [
+                        {self.mgmt_details[machine_name]['mgmt_username']},
+                        {self.mgmt_details[machine_name]['mgmt_password']},
+                    ]
+                    run_cmd(cmd=cmd, secrets=secrets)
+                    # Changes boot prioriy to pxe
+                    cmd = f"ipmitool -I lanplus -U {self.mgmt_details[machine_name]['mgmt_username']} " \
+                          f"-P {self.mgmt_details[machine_name]['mgmt_password']} " \
+                          f"-H {self.mgmt_details[machine_name]['mgmt_console']} chassis bootdev pxe"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:230]: same

File: ocs_ci/deployment/baremetal.py [hunk 8]
```diff
@@ -1,13 +1,410 @@
...
+                    run_cmd(cmd=cmd, secrets=secrets)
+                    # Power On Machine
+                    cmd = f"ipmitool -I lanplus -U {self.mgmt_details[machine_name]['mgmt_username']} " \
+                          f"-P {self.mgmt_details[machine_name]['mgmt_password']} " \
+                          f"-H {self.mgmt_details[machine_name]['mgmt_console']} chassis power on"
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:235]: same

File: ocs_ci/deployment/baremetal.py [hunk 9]
```diff
@@ -1,13 +1,410 @@
...
+            f"-- chroot /host lsblk -nd -e252,7 --output NAME --json"
+        )
+        out = ocp_obj.exec_oc_cmd(
+            command=cmd, out_yaml_format=False,
+        )
+        lsblk_output = json.loads(str(out))
+        lsblk_devices = lsblk_output['blockdevices']
+        for lsblk_device in lsblk_devices:
+            base_cmd = """pvs --config "devices{filter = [ 'a|/dev/%s.*|', 'r|.*|' ] }" --reportformat json""" \
+                       % lsblk_device['name']
```
Comment (ebenahar) [inline ocs_ci/deployment/baremetal.py:359]: break lines with parentheses

**Comment by prsurve** [comment]
Closing BZ will create a new one soon

**Pull request closed without merging**